### ISSUE TITLE
 ModuleNotFoundError: No module named 'fixBinaries'

### ISSUE THREAD

**Issue by fuxiocteract** [coin-or/CyLP#74]
ModuleNotFoundError: No module named 'fixBinaries'

Install on macOS 14 beta5 with python 3

With command pip install --pre cylp, cbc has already been installed.

**Comment by fuxiocteract** [comment]
Looking in indexes: https://pypi.python.org/simple, https://pypi.apple.com/simple
Collecting cylp
  Using cached https://files.pythonhosted.org/packages/00/d0/b0a8ba0e26c2ea4c2882e5102648e2438cdf39b213dc8b9556c6d586e6ab/cylp-0.9.0b1.tar.gz
Requirement already satisfied: numpy>=1.5.0 in ./venv/lib/python3.7/site-packages (from cylp) (1.17.0)
Requirement already satisfied: scipy>=0.10.0 in ./venv/lib/python3.7/site-packages (from cylp) (1.3.0)
Installing collected packages: cylp
  Running setup.py install for cylp ... error
    Complete output from command /Users/shuohuang/Projects/algorithm-comparison/venv/bin/python -u -c "import setuptools, tokenize;__file__='/private/var/folders/1v/n61gzjhs7lq5ssym67d6cjj00000gp/T/pip-install-vj_9z5d1/cylp/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /private/var/folders/1v/n61gzjhs7lq5ssym67d6cjj00000gp/T/pip-record-svrng_sr/install-record.txt --single-version-externally-managed --compile --install-headers /Users/shuohuang/Projects/algorithm-comparison/venv/include/site/python3.7/cylp:
    running install
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/1v/n61gzjhs7lq5ssym67d6cjj00000gp/T/pip-install-vj_9z5d1/cylp/setup.py", line 417, in <module>
        package_data={"cylp": extra_files})
      File "/Users/shuohuang/Projects/algorithm-comparison/venv/lib/python3.7/site-packages/setuptools-40.8.0-py3.7.egg/setuptools/__init__.py", line 145, in setup
      File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/dist.py", line 966, in run_commands
        self.run_command(cmd)
      File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/private/var/folders/1v/n61gzjhs7lq5ssym67d6cjj00000gp/T/pip-install-vj_9z5d1/cylp/setup.py", line 389, in run
        from fixBinaries import platform_dir
    ModuleNotFoundError: No module named 'fixBinaries'
    
    ----------------------------------------
Command "/Users/shuohuang/Projects/algorithm-comparison/venv/bin/python -u -c "import setuptools, tokenize;__file__='/private/var/folders/1v/n61gzjhs7lq5ssym67d6cjj00000gp/T/pip-install-vj_9z5d1/cylp/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /private/var/folders/1v/n61gzjhs7lq5ssym67d6cjj00000gp/T/pip-record-svrng_sr/install-record.txt --single-version-externally-managed --compile --install-headers /Users/shuohuang/Projects/algorithm-comparison/venv/include/site/python3.7/cylp" failed with error code 1 in /private/var/folders/1v/n61gzjhs7lq5ssym67d6cjj00000gp/T/pip-install-vj_9z5d1/cylp/

**Comment by fuxiocteract** [comment]
If I install using pip install cylp I got this:

equirement already satisfied: numpy>=1.5.0 in /Users/shuohuang/Projects/algorithm-comparison/venv/lib/python3.7/site-packages (from cylp==0.9.0b1) (1.17.0)
Requirement already satisfied: scipy>=0.10.0 in /Users/shuohuang/Projects/algorithm-comparison/venv/lib/python3.7/site-packages (from cylp==0.9.0b1) (1.3.0)
Installing collected packages: cylp
  Running setup.py develop for cylp
    Complete output from command /Users/shuohuang/Projects/algorithm-comparison/venv/bin/python -c "import setuptools, tokenize;__file__='/Users/shuohuang/Downloads/cylp-0.9.0b1/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" develop --no-deps:
    running develop
    running egg_info
    writing cylp.egg-info/PKG-INFO
    writing dependency_links to cylp.egg-info/dependency_links.txt
    writing requirements to cylp.egg-info/requires.txt
    writing top-level names to cylp.egg-info/top_level.txt
    reading manifest file 'cylp.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    writing manifest file 'cylp.egg-info/SOURCES.txt'
    running build_ext
    building 'cylp.cy.CyClpPrimalColumnPivotBase' extension
    gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch x86_64 -g -I./cylp/cpp -I./cylp/cy -I/usr/local/Cellar/cbc/2.10.2_1/include/coin -I/Users/shuohuang/Projects/algorithm-comparison/venv/lib/python3.7/site-packages/numpy/core/include -I. -I/Users/shuohuang/Projects/algorithm-comparison/venv/include -I/Library/Frameworks/Python.framework/Versions/3.7/include/python3.7m -c cylp/cpp/IClpPrimalColumnPivotBase.cpp -o build/temp.macosx-10.9-x86_64-3.7/cylp/cpp/IClpPrimalColumnPivotBase.o -w
    In file included from cylp/cpp/IClpPrimalColumnPivotBase.cpp:1:
    ./cylp/cpp/IClpPrimalColumnPivotBase.h:5:10: fatal error: 'ClpPrimalColumnPivot.hpp' file not found
    #include "ClpPrimalColumnPivot.hpp"
             ^~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
    error: command 'gcc' failed with exit status 1
    
    ----------------------------------------
Command "/Users/shuohuang/Projects/algorithm-comparison/venv/bin/python -c "import setuptools, tokenize;__file__='/Users/shuohuang/Downloads/cylp-0.9.0b1/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" develop --no-deps" failed with error code 1 in /Users/shuohuang/Downloads/cylp-0.9.0b1/

**Comment by tkralphs** [comment]
`pip install cylp` is not going to work, since this gets you the old Python 2 version. As far as what's going on with `pip install --pre cylp`, The missing module `fixBinaries.py` is in the repository in the same directory as `setup.py`, but I guess it is not being found somehow when installing from Pypi. Can you try just cloning `master` and then doing `pip install .` instead? I'm not entirely sure what all the OS X-related fixes are doing in the setup script.

**Comment by tkralphs** [comment]
fixBinaries is no longer used, so closing this bug.

**Comment by Rothdyt** [comment]
When I try to build from source, I obtained the following error messages:

```
    gcc -pthread -B /home/dyt/anaconda3/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/usr/include/coin-or -I./cylp/cpp -I./cylp/cy -I/home/dyt/anaconda3/lib/python3.7/site-packages/numpy/core/include -I. -I/usr/include/coin -I/home/dyt/anaconda3/include/python3.7m -c cylp/cpp/IClpPrimalColumnPivotBase.cpp -o build/temp.linux-x86_64-3.7/cylp/cpp/IClpPrimalColumnPivotBase.o -w -std=c++11
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
    In file included from cylp/cpp/IClpPrimalColumnPivotBase.cpp:1:0:
    cylp/cpp/IClpPrimalColumnPivotBase.h:5:10: fatal error: ClpPrimalColumnPivot.hpp: No such file or directory
     #include "ClpPrimalColumnPivot.hpp"
              ^~~~~~~~~~~~~~~~~~~~~~~~~~
    compilation terminated.
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/dyt/anaconda3/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-n_mvz3qd/cylp/setup.py'"'"'; __file__='"'"'/tmp/pip-install-n_mvz3qd/cylp/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-z_b16qg5/install-record.txt --single-version-externally-managed --compile --install-headers /home/dyt/anaconda3/include/python3.7m/cylp Check the logs for full command output.
```

Indeed, the missing file can be found at

```
cylp/cy/ClpPrimalColumnPivot.hpp
```

I do not know how to address this issue.

**Comment by tkralphs** [comment]
Can you open a new issue, as this seems to be unrelated to the original one? Give as much detail as you can. It seems you are using the Anaconda. I have never tried using Anaconda on OS X. Homebrew Python is probably a better bet.

**Comment by dasguptap** [comment]
The problem of  "ClpPrimalColumnPivot.hpp: No such file or directory" can be fixed by replacing the line below (line 86 in setup.py):
 incDirs = [join(CoinDir, 'include', 'coin')]
with this line:
 incDirs = [join(CoinDir, 'include', 'coin-or')]

In the latest version 2.10.5 of coin-or, the include files are under <install dir>/include/coin-or

**Comment by tkralphs** [comment]
This was just fixed in 6904134bb81e4
